### ISSUE TITLE
Change signature method 

### ISSUE THREAD

**Issue by duydao** [simov/grant#183]
Hi,

I'm trying to conncet to our local jira/confluence servers using Grant. There seems to be no way to change the signature method to RSA-SHA1, which is required by the oauth 1.0a servers.

I've managed to modify the oauth1.js in order to add support for the custo provider. Is there another way to achieve this without messing with the npm package files?

**Comment by simov** [comment]
No, that's not exposed, but it looks like support for it should be straightforward to add, because Grant already accepts a `private_key` configuration key, so probably switching to `RSA-SHA1` in that case would solve it.

I'll try to get the Jira integration working over the weekend, and I'll let you know.

**Comment by simov** [comment]
Fixed here https://github.com/simov/grant/commit/d83126dace86270ca572a55dd9b9f56c694f669a, no tests yet, but it works.

Here is an example configuration:

```json
{
  "applinks": {
    "request_url": "https://[subdomain].atlassian.net/plugins/servlet/oauth/request-token",
    "authorize_url": "https://[subdomain].atlassian.net/plugins/servlet/oauth/authorize",
    "access_url": "https://[subdomain].atlassian.net/plugins/servlet/oauth/access-token",
    "oauth": 1,
    "subdomain": "your instance",
    "key": "OauthKey",
    "private_key": "-----BEGIN PRIVATE KEY-----\n...actual key here...\n-----END PRIVATE KEY-----\n\n"
  }
}
```

Note that you can pass your `private_key` as raw Buffer using the [dynamic state](https://github.com/simov/grant#dynamic-state), here is an [example](https://github.com/simov/grant/tree/master/examples/dynamic-state).

This will go in the next release, but in the meantime you'll have to set your dependency version as `github:simov/grant` in package.json. I'll let you know when it's published.

**Comment by duydao** [comment]
Youre awesome, thanks alot @simov

**Comment by simov** [comment]
Just published v5.3.0 with the fix :tada:
